src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,48 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+
+    private K[] keyArray = (K[]) new Object[10];
+    private V[] valueArray = (V[]) new Object[10];
+    private int freeIndex = 0;
+
     @Override
     public void put(K key, V value) {
+        int i = 0;
+        for (; i < freeIndex; i++) {
+            if ((keyArray[i] != null
+                    && keyArray[i].equals(key))
+                    || (keyArray[i] == null
+                    && keyArray[i] == key)) {
```

Comment:
```suggestion
            if ((keyArray[i] != null  && keyArray[i].equals(key))  || keyArray[i] == key) {
```
Цю умову можна спростити, якщо у нас дві змінні мають значення null, то їх можна порівняти через first == second, інші випадки опрацює умова  keyArray[i] != null  && keyArray[i].equals(key)